File: Python/Data Structures and Algorithms/Linear Shoal/ArrayI/MaxConsecutiveOnes.py
""" Given a binary array nums, return the maximum number of consecutive 1's in the array. """

class Solution:
    def findMaxConsecutiveOnes(self, nums: list[int]) -> int:
        greaterConsecutive = 0
        current_sum = 0

        for x in nums:
            if x == 1:
                current_sum += 1
            elif x == 0:
                if greaterConsecutive < current_sum:
                    greaterConsecutive = current_sum 
                
                current_sum = 0

        return max(greaterConsecutive, current_sum)

File: Cpp/Data Structures and Algorithms/Linear Shoal/Array I/MaxConsecutiveOnes.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int total = 0, maxTotal = 0;

        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] == 0) {
                total = 0;
            } else {
                total++;
            }

            maxTotal = max(total, maxTotal);
        }
        
        maxTotal = max(total, maxTotal);
        
        return maxTotal;
    }
};